.github/workflows/ci.yml

```diff
@@ -46,6 +50,59 @@ jobs:
     - name: Install dependencies
       run: make setup
 
+    - name: Install Deno Jupyter Kernel
+      run: deno jupyter --install
+
+    - name: Fix deno path on macOS
+      if: runner.os == 'macOS'
+      run: |
+        source .venv/bin/activate
+        export JUPYTER_PLATFORM_DIRS=0
+
+        # ensure jupyter can find the kernel config using the old path structure
+        jupyter kernelspec list | grep -q deno || exit 1
+
+        ORIG_JUPYTER_KERNEL_PATH="$(jupyter --data-dir)/kernels"
+        
+        export JUPYTER_PLATFORM_DIRS=1
+
+        # make the data directory for the new path structure
+        mkdir -p "$(jupyter --data-dir)/kernels"
+
+        cp -r "$ORIG_JUPYTER_KERNEL_PATH/deno" "$(jupyter --data-dir)/kernels"
+
+        # fail if jupyter still can't find the kernel config
+        jupyter kernelspec list | grep -q deno
+        
+        deactivate
+
+    - name: Fix deno path on Windows
+      if: runner.os == 'Windows'
+      shell: pwsh
+      run: |
+        .\.venv\Scripts\Activate.ps1
+        $env:JUPYTER_PLATFORM_DIRS = "0"
+
+        # ensure jupyter can find the kernel config using the old path structure
+        $kernelCheck = jupyter kernelspec list | Select-String -Pattern "deno"
+        if (-not $kernelCheck) { exit 1 }
+
+        $origKernelPath = Join-Path (jupyter --data-dir) "kernels"
+        
+        $env:JUPYTER_PLATFORM_DIRS = "1"
+
+        # make the data directory for the new path structure
+        $newKernelPath = Join-Path (jupyter --data-dir) "kernels"
+        New-Item -ItemType Directory -Force -Path $newKernelPath
+
+        Copy-Item -Path (Join-Path $origKernelPath "deno") -Destination $newKernelPath -Recurse -Force
+
+        # fail if jupyter still can't find the kernel config
+        $kernelCheck = jupyter kernelspec list | Select-String -Pattern "deno"
+        if (-not $kernelCheck) { exit 1 }
+        
+        deactivate
+
```

Comment:
issue (ci perf): We're in the `quality` job, which only runs on ubuntu-latest, so I don't think we need these.